web-ui/src/components/reports-section/checkin-report/checkin-utils.test.js

```diff
@@ -65,58 +65,93 @@ describe('getLastCheckinDate', () => {
+      { checkInDate: [2024, 6, 30, 14, 30] } // June 30, 2024
     ];
 
-    const result = getCheckinDateForPeriod(checkins, mockReportDate);
+    const result = getCheckinDateForPeriod(checkins, mockReportDateQ2);
 
     expect(result).toBeInstanceOf(Date);
     expect(result.getFullYear()).toBe(2024);
-    expect(result.getMonth()).toBe(4); // May is 4 (zero-indexed)
-    expect(result.getDate()).toBe(1); // Latest date in the array
+    expect(result.getMonth()).toBe(5); // June is 5 (zero-indexed)
+    expect(result.getDate()).toBe(30); // Latest date in the array
     expect(result.getHours()).toBe(14);
     expect(result.getMinutes()).toBe(30);
   });
 
-  test('returns correct date regardless of checkin status (completed or not)', () => {
+  test('returns correct date regardless of checkin status (completed or not) for Q1', () => {
     const checkins = [
-      { checkInDate: [2024, 3, 1, 10, 0], completed: true }, // March 1, 2024 (completed)
-      { checkInDate: [2024, 4, 1, 9, 0], completed: false }, // April 1, 2024 (not completed)
-      { checkInDate: [2024, 4, 15, 14, 30], completed: false } // April 15, 2024 (not completed)
+      { checkInDate: [2024, 1, 10, 10, 0], completed: true }, // January 10, 2024 (completed)
+      { checkInDate: [2024, 2, 20, 9, 0], completed: false }, // February 20, 2024 (not completed)
+      { checkInDate: [2024, 3, 31, 14, 30], completed: false } // March 31, 2024 (not completed)
     ];
 
-    const result = getCheckinDateForPeriod(checkins, mockReportDate);
+    const result = getCheckinDateForPeriod(checkins, mockReportDateQ1);
 
     expect(result).toBeInstanceOf(Date);
     expect(result.getFullYear()).toBe(2024);
-    expect(result.getMonth()).toBe(3); // April is 3 (zero-indexed)
-    expect(result.getDate()).toBe(15); // Latest date in the array
+    expect(result.getMonth()).toBe(2); // March is 2 (zero-indexed)
+    expect(result.getDate()).toBe(31); // Latest date in the array
     expect(result.getHours()).toBe(14);
     expect(result.getMinutes()).toBe(30);
   });
+
+  test('returns correct date regardless of checkin status (completed or not) for Q2', () => {
+    const checkins = [
+      { checkInDate: [2024, 4, 10, 10, 0], completed: true }, // April 10, 2024 (completed)
+      { checkInDate: [2024, 5, 20, 9, 0], completed: false }, // May 20, 2024 (not completed)
+      { checkInDate: [2024, 6, 30, 14, 30], completed: false } // June 30, 2024 (not completed)
+    ];
+
+    const result = getCheckinDateForPeriod(checkins, mockReportDateQ2);
+
+    expect(result).toBeInstanceOf(Date);
+    expect(result.getFullYear()).toBe(2024);
+    expect(result.getMonth()).toBe(5); // June is 5 (zero-indexed)
+    expect(result.getDate()).toBe(30); // Latest date in the array
+    expect(result.getHours()).toBe(14);
+    expect(result.getMinutes()).toBe(30);
+  });
+
+  test('returns correct date when check-ins are on Feb 12 and May 1 with a report date in Q1', () => {
```

Comment:
Under previous logic this test would return 5/1 as the Q1 check-in date unexpectedly.